### PR TITLE
feat(privval)!: Allow Privval to sign arbitrary bytes.

### PR DESCRIPTION
**Pull request by itsdevbear**
Closes https://github.com/cometbft/cometbft/issues/2517

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by faddat** [comment]
saucy!

**Comment by cason** [comment]
Do you mind adding some test units covering the added feature?

**Comment by itsdevbear** [comment]
@cason no problem

File: proto/cometbft/privval/v1/types.proto [hunk 1]
```diff
@@ -49,6 +49,17 @@ message SignedProposalResponse {
   RemoteSignerError          error    = 2;
 }
 
+// SignBytesRequest is a request to sign arbitrary bytes
+message SignBytesRequest {
+  bytes  value       = 1;
```
Comment (cason) [inline proto/cometbft/privval/v1/types.proto:54]:
```suggestion
  bytes  value = 1;
```

File: proto/cometbft/privval/v1/types.proto [hunk 2]
```diff
@@ -49,6 +49,17 @@ message SignedProposalResponse {
...
+}
+
+// SignBytesResponse is a response containing a signature or an error
+message SignBytesResponse {
+  bytes             signature = 1;
+  RemoteSignerError         error    = 2;
```
Comment (cason) [inline proto/cometbft/privval/v1/types.proto:60]:
```suggestion
  RemoteSignerError error     = 2;
```

**Review by cason — APPROVED** [review]
Conceptually and implementation-wise legit.

But I wonder whether we should somehow limit the maximum size of the array of bytes that can be signed. As this can constitute an attack vector.

**Comment by itsdevbear** [comment]
@cason good with all those changes,

re: max length, I mean I suppose? but feels overkill imo.

**Comment by cason** [comment]
Maybe is too much establishing a maximum size for the data that can be send to the remote signer.

But I would like to have a second opinion from the team.

**Comment by cason** [comment]
I assume this closes https://github.com/cometbft/cometbft/issues/2517?

**Comment by itsdevbear** [comment]
ya

**Comment by adizere** [comment]
@itsdevbear is this needed on v0.38, or is it enough to backport to v1 ?

**Comment by itsdevbear** [comment]
@adizere both would be great, or get the sdk onto v1 :P

**Review by melekes — APPROVED** [review]
Thanks @itsdevbear ❤️

**Comment by melekes** [comment]
@itsdevbear would you mind adding a changelog entry?

> But I wonder whether we should somehow limit the maximum size of the array of bytes that can be signed. As this can constitute an attack vector.

The remote signer would be responsible for having a hard limit, after which it would refuse to sign bytes.

> is this needed on v0.38, or is it enough to backport to v1 ?

How could we backport this if it's a breaking change?

**Comment by cason** [comment]
Can we merge this PR?

Could the discussion about backporting it to v0.38 proceed on the associated issue https://github.com/cometbft/cometbft/issues/2517?

**Comment by cason** [comment]
> @itsdevbear would you mind adding a changelog entry?

Ah, we are also missing this.

**Comment by itsdevbear** [comment]
@cason done

**Comment by melekes** [comment]
The changelog entry must be added to `.changelog`, not `CHANGELOG.md`.

**Pull request merged**